packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -544,118 +542,107 @@ export function NativeStackView({ state, navigation, descriptors }: Props) {
 
   const currentRouteKey = state.routes[state.index].key;
   const topScreenOptions = descriptors[currentRouteKey].options;
-  const gestureDetectorBridge = React.useRef<GestureDetectorBridge>({
-    stackUseEffectCallback: (_stackRef) => {
-      // this method will be overriden in GestureDetector
-    },
-  });
 
   const screensRefs = React.useRef<ScreensRefsHolder>({});
 
-  const ScreenGestureDetector = React.useContext(GHContext);
-
   useInvalidPreventRemoveError(descriptors);
 
   const modalRouteKeys = getModalRouteKeys(state.routes, descriptors);
 
   return (
     <SafeAreaProviderCompat style={{ backgroundColor: colors.background }}>
-      <ScreenGestureDetector
-        gestureDetectorBridge={gestureDetectorBridge}
+      <ScreenStack
+        style={styles.container}
+        /** Props for custom screen transitions */
         goBackGesture={topScreenOptions?.gestureType}
         transitionAnimation={topScreenOptions?.animationForGesture}
         screenEdgeGesture={topScreenOptions?.gestureFromEdgeEnabled ?? false}
         screensRefs={screensRefs}
         currentRouteKey={currentRouteKey}
```

Comment:
maybe better to name this prop as `currentId` or something like that since `route.key` is a react navigation specific concept, so may not be appropriate for screens

Reply:
changed in: https://github.com/react-navigation/react-navigation/pull/12204/commits/057e143e691c16f7a99edc622cf764a5ba3769a6